vp/workflow/views.py

```diff
@@ -230,14 +233,40 @@ def upload_file(request):
     if f is None:
         return JsonResponse("Empty content", status=404)
 
-    node_id = request.POST.get('nodeId', '')
-
     try:
-        save_name = request.pyworkflow.upload_file(f, node_id)
-        return JsonResponse({"filename": save_name}, status=201, safe=False)
+        node_id = request.POST.get('nodeId')
+
+        if node_id is None:
+            # custom node file
+            file_path = request.pyworkflow.node_path('custom_node', f.name)
```

Comment:
This should be `custom_nodes`. Also it either needs to exist in the repo, or be dynamically created, or this line will failed when called from the front end.

Reply:
Oh wait I see your last commit does create it. So just fixing the typo here should do the trick!